### DISCUSSION TITLE
Form Validation - Payments - Determining type of card entered

### DISCUSSION THREAD

**Discussion by jike212** [fomantic/Fomantic-UI#1904] [Behaviors (Api, Form Validation, Visibility)] [answered]
I'm putting together a form with validation to ensure the user inputs specific card types (Visa, Mastercard, Amex, etc.) but I would also like to detect what kind of card the user has input once it's been deemed valid.

I'm using the documentation here: https://fomantic-ui.com/behaviors/form.html#payment

I just can't seem to find a way to extract what kind of card it is from Fomantic once it passes the validation. Any ideas?

**Comment by lubber-de** [comment, marked as answer]
You may have to check the regexes from the validation function. I prepared a little example which does what you need. Feel free to adopt and change it according to your needs
https://jsfiddle.net/lubber/hzt017q3/13/

![image](https://user-images.githubusercontent.com/18379884/111908646-42f56400-8a5a-11eb-8a50-50cd117efd1a.png)

**Reply by jike212** [reply]
Thank you @lubber-de, I actually ended up using the `cleave.js` library as it added input masking as well. I submitted a PR for the issue regarding card numbers with spaces not validating. Thanks for approving that PR and hopefully it gets one more approval so it can be implemented in the main library for others!
